osx/OSXMain.mm

```diff
@@ -126,64 +191,59 @@ void BeginOSX()
             continue;
         }
 
-        NSString *pStrSaveFilePath = [pStrCaseSavesFilePath stringByAppendingPathComponent:pStrSaveFileName];
-		ppSaveFilePathList.push_back(string([pStrSaveFilePath fileSystemRepresentation]));
+        NSString *pStrSaveFilePath = [currentCaseSavePath stringByAppendingPathComponent:pStrSaveFileName];
+        ppSaveFilePathList.push_back(string([pStrSaveFilePath fileSystemRepresentation]));
     }
 
-    [pool drain];
+    AUTORELEASE_POOL_STOP
     return ppSaveFilePathList;
 }
 
-string GetVersionStringOSX(string PropertyListFilePath)
+string GetVersionStringOSX()
 {
-    NSFileManager *defaultManager = [NSFileManager defaultManager];
-    NSString *pProperyListPath = [defaultManager stringWithFileSystemRepresentation:PropertyListFilePath.c_str() length: PropertyListFilePath.size()];
-
-    if (![defaultManager fileExistsAtPath:pProperyListPath])
-    {
-        return string();
-    }
-
-    NSDictionary *pPropertyListDictionary =
-        [NSDictionary dictionaryWithContentsOfFile:pProperyListPath];
-
-    if (pPropertyListDictionary == NULL)
-    {
-        return string();
-    }
-
-    NSString *pVersionString = [pPropertyListDictionary objectForKey:@"VersionString"];
-    return [pVersionString UTF8String];
+    AUTORELEASE_POOL_START
+    NSBundle *ourBundle = [NSBundle mainBundle];
+    NSDictionary *infoPlist = ourBundle.infoDictionary;
+    NSString *versStr = [infoPlist objectForKey:@"CFBundleVersion"];
+    return versStr != nil ? versStr.UTF8String : "";
+    AUTORELEASE_POOL_STOP
 }
 
-char * GetPropertyListXMLForVersionStringOSX(string pPropertyListFilePath, string pVersionString, unsigned long *pVersionStringLength)
+char *GetPropertyListXMLForVersionStringOSX(const string &pPropertyListFilePath, const string &pVersionString, unsigned long *pVersionStringLength)
 {
+    AUTORELEASE_POOL_START
     *pVersionStringLength = 0;
 
     NSFileManager *defaultManager = [NSFileManager defaultManager];
     NSString *pErrorDesc = nil;
     //TODO: Save the NSString as, say, a static pointer.
-    NSString *pProperyListPath = [defaultManager stringWithFileSystemRepresentation:pPropertyListFilePath.c_str() length: pPropertyListFilePath.size()];
+    NSString *pProperyListPath = [defaultManager
```

Comment:
Not your mistake, but while you're here, might as well fix my typo of "propery" (should be "property").